lib/internal/util.js

```diff
@@ -366,6 +366,39 @@ function spliceOne(list, index) {
   list.pop();
 }
 
+function isInsideNodeModules() {
+  // Lazy-load to avoid a circular dependency.
+  const { runInNewContext } = require('vm');
+  // Use `runInNewContext()` to get something tamper-proof and side-effect-free.
+  // Since this is currently only used for a deprecated API, at most once
+  // per process, the perf implications should be okay.
```

Comment:
The problem here is that this is _not_ called once per process. It's called once per every `new Buffer()` call until it finds a one that is not inside `node_process`.

How long would 1k / 10k calls take?

Perhaps, bail out after 1k/10k per process if no other suggestions?

Reply:
Right – I switched to caching the function now.

I get ~ 16µs per call for this, so 10k calls would be 0.16 seconds.

(That’s a lot, yes.)